python/base_computer/ship_view.py

```diff
@@ -0,0 +1,304 @@
+        
+    return text
+  
+def get_durability(ship_stats):
+    armor = [
+        ('Fore-starboard-high','Armor_Front_Top_Right'),
+        ('Aft-starboard-high','Armor_Back_Top_Right'),
+        ('Fore-port-high','Armor_Front_Top_Left'),
+        ('Aft-port-high','Armor_Back_Top_Left'),
+        ('Fore-starboard-low','Armor_Front_Bottom_Right'),
+        ('Aft-starboard-low','Armor_Back_Bottom_Right'),
+        ('Fore-port-low','Armor_Front_Bottom_Left'),
+        ('Aft-port-low','Armor_Back_Bottom_Left'),
+    ]
+    
+    shield4 = [
+        ('Port','Shield_Front_Bottom_Left'),('Starboard','Shield_Front_Bottom_Right'),('Fore','Shield_Front_Top_Right'),('Aft','Shield_Back_Top_Left'),
+    ]
+    
+    shield2 = [
+        ('Fore','Shield_Front_Top_Right'),('Aft','Shield_Back_Top_Left'),
+    ]
+    
+    # This is a kludge. Should go away when we refactor units.json and unit_csv.cpp
+    shield_stat = {}
+    num_emitters = 0
+    for pair in shield4:
+        if ship_stats[pair[1]] == '':
+            continue
+        value = get_dbl(ship_stats,pair[1])
+        if value > 0:
+            num_emitters += 1
+    
+            
+    hull = lnum(ship_stats,'Hull')
+    shield_recharge = lnum(ship_stats,'Shield_Recharge')
+    
+    text = f"{green}[DURABILITY STATISTICS]{newline}#-c{newline}"
+    text += f"{light_grey}Sustainable Hull Damage: #-c{hull} MJ{newline}{newline}"
+    
+    text += f"{light_grey}Armor#-c{newline}"
+    for pair in armor:
+        armor_stat = lnum(ship_stats, pair[1])
+        text += f"{light_yellow} - {pair[0]}: #-c{armor_stat} MJ{newline}"
+    
+    text += f"{newline}{light_grey}Shields#-c{newline}"
+    text += f"{light_grey}Shield recharge: #-c{shield_recharge} MJ/s{newline}"
+    if num_emitters == 0:
+        text += f"{red}No shielding.{end_color}{newline}{newline}"
+    elif num_emitters == 2:
+        text += f"{light_grey}Number of shield emitter: 2{end_color}{newline}"
+        for pair in shield2:
+            shield_stat = lnum(ship_stats, pair[1])
+            text += f"{light_yellow} - {pair[0]}: #-c{shield_stat} MJ{newline}"
+    else:
+        text += f"{light_grey}Number of shield emitter: 4{end_color}{newline}"
+        for pair in shield4:
+            shield_stat = lnum(ship_stats, pair[1])
+            text += f"{light_yellow} - {pair[0]}: #-c{shield_stat} MJ{newline}"
+        
+    return text
+
+
+    
+def get_ship_description(ship_stats):
+    locale.setlocale(locale.LC_ALL, '')
+    
+    text = get_notes(ship_stats)
+    text += get_general(ship_stats)
+    text += get_flight(ship_stats)
+    text += get_governor(ship_stats)
+    text += get_radar(ship_stats)
+    text += get_energy_spec_and_jump(ship_stats)
+    text += get_durability(ship_stats)
+                      
+    return text
+
+# Test function
+# Useful to check python script for correctness before running VS
+
+t = get_ship_description(llama_begin)
+t = t.replace('#n#','\n')
+print(t)
```

Comment:
if this is for testing, then it probably should be wrapped. Usually if its a test to be called on the commandline (e.g just see the script output) then using:
```python
if __name__ == "__main__":
   # python block
```
would be appropriate. If we can control the name and calling it in a test, then may be we call it `__vegastrike_test_main__`  instead...but I doubt that'd we'd have that control.